modules/util.py

```diff
@@ -773,19 +789,23 @@ def nohardlink(self, file, notify, ignore_root_dir):
         This fixes the bug in #192
         """
 
-        def has_hardlinks(self, file, ignore_root_dir):
+        def has_hardlinks(self, file, ignore_root_dir, category, ignore_category_dir):
             """
             Check if a file has hard links.
 
             Args:
                 file (str): The path to the file.
                 ignore_root_dir (bool): Whether to ignore the root directory.
+                category (str): The category name.
+                ignore_category_dir (bool): Whether to ignore the torrent category directory.
 
             Returns:
                 bool: True if the file has hard links, False otherwise.
             """
             if ignore_root_dir:
-                return os.stat(file).st_nlink - self.inode_count.get(os.stat(file).st_ino, 1) > 0
+                return os.stat(file).st_nlink - self.root_inode_count.get(os.stat(file).st_ino, 1) > 0
+            elif ignore_category_dir:
+                return os.stat(file).st_nlink - self.categories_inode_count.get(category, {}).get(os.stat(file).st_ino, 1) > 0
```

Comment:
Has this logic been thoroughly tested? Could you provide some trace log output for specific examples to show it's working as intended?